src/strategy.rs

```diff
@@ -275,29 +278,225 @@ pub fn hostname<S: BuildHasher>(parameters: Option<HashMap<String, String, S>>)
 }
 
 /// returns true if the strategy should be delegated to, false to disable
-fn _compile_constraint_string<F>(expression: ConstraintExpression, getter: F) -> Evaluate
+fn _compile_constraint_string<F>(
+    expression: ConstraintExpression,
+    inverted: Option<bool>,
+    getter: F,
+) -> Evaluate
 where
     F: Fn(&Context) -> Option<&String> + Clone + Sync + Send + 'static,
 {
     match &expression {
```

Comment:
This match is getting a little bit long. I'm struggling to break it out into more granular functions due to rust marking closures as unique types. I'm very open to suggestions if someone feels the same as me and has clever ideas here